dChatServer/ChatPacketHandler.cpp

```diff
@@ -337,121 +337,142 @@ void ChatPacketHandler::HandleRemoveFriend(Packet* packet) {
+			bitStream.Write<uint32_t>(0); // not used
+			bitStream.Write(LUWString(sender.playerName));
+			bitStream.Write(sender.playerID);
+			bitStream.Write<uint16_t>(0); // sourceID
+			bitStream.Write(sender.GMLevel);
+			bitStream.Write(LUWString(otherMember.playerName));
+			bitStream.Write(otherMember.GMLevel);
+			bitStream.Write(eChatMessageResponseCode::Sent);
+			bitStream.Write(message);
+
+			SystemAddress sysAddr = otherMember.sysAddr;
+			SEND_PACKET;
+		}
+	}
+}
 
-	for (const auto memberId : team->memberIDs) {
-		const auto& otherMember = Game::playerContainer.GetPlayerData(memberId);
+void ChatPacketHandler::HandlePrivateChatMessage(Packet* packet) {
+	CINSTREAM_SKIP_HEADER;
+	LWOOBJID playerID;
+	inStream.Read(playerID);
 
-		if (!otherMember) return;
+	const auto& sender = Game::playerContainer.GetPlayerData(playerID);
+	if (!sender || sender.GetIsMuted()) return;
 
-		CBITSTREAM;
-		BitStreamUtils::WriteHeader(bitStream, eConnectionType::CHAT_INTERNAL, eChatInternalMessageType::ROUTE_TO_PLAYER);
-		bitStream.Write(otherMember.playerID);
+	eChatChannel channel;
+	uint32_t size;
+	LUWString receiverName;
 
-		BitStreamUtils::WriteHeader(bitStream, eConnectionType::CHAT, eChatMessageType::PRIVATE_CHAT_MESSAGE);
-		bitStream.Write(otherMember.playerID);
-		bitStream.Write<uint8_t>(8);
-		bitStream.Write<unsigned int>(69);
-		bitStream.Write(LUWString(sender.playerName));
-		bitStream.Write(sender.playerID);
-		bitStream.Write<uint16_t>(0);
-		bitStream.Write<uint8_t>(0); //not mythran nametag
-		bitStream.Write(LUWString(otherMember.playerName));
-		bitStream.Write<uint8_t>(0); //not mythran for receiver
-		bitStream.Write<uint8_t>(0); //teams?
-		bitStream.Write(LUWString(message, 512));
-
-		SystemAddress sysAddr = otherMember.sysAddr;
-		SEND_PACKET;
-	}
-}
+	inStream.IgnoreBytes(4);
+	inStream.Read(channel);
+	inStream.Read(size);
+	inStream.IgnoreBytes(77);
+	inStream.Read(receiverName);
+	inStream.IgnoreBytes(2);
 
-void ChatPacketHandler::HandlePrivateChatMessage(Packet* packet) {
-	LWOOBJID senderID = PacketUtils::ReadS64(0x08, packet);
-	std::string receiverName = PacketUtils::ReadString(0x66, packet, true);
-	std::string message = PacketUtils::ReadString(0xAA, packet, true, 512);
+	LUWString message(size);
+	inStream.Read(message);
+	
+	LOG("Got a message from (%s) via [%s]: %s to %s", sender.playerName.c_str(), StringifiedEnum::ToString(channel).data(), message.GetAsString().c_str(), receiverName.GetAsString().c_str());
 
-	//Get the bois:
-	const auto& goonA = Game::playerContainer.GetPlayerData(senderID);
-	const auto& goonB = Game::playerContainer.GetPlayerData(receiverName);
-	if (!goonA || !goonB) return;
+	const auto& receiver = Game::playerContainer.GetPlayerData(receiverName.GetAsString());
+	if (!receiver) return; // TODO: respond with not online or error
 
-	if (goonA.GetIsMuted()) return;
+	// TODO: check if they are friends, only friends should be able to whisper/private chat
```

Comment:
container has this data already.  Why was the muted check removed?

Reply:
mute check is done above, was a redundant check (will double check tho)

Reply:
Also just didn't do it yet cause I wanted to keep this PR from getting to big, but can do

Reply:
yeah, goona is sender and the sender is check if muted above, so redundant check

Reply:
the friends check isnt even much more code
```
bool isFriends = std::find(friends.begin(), friends.end(), friendId) != friends.end();
if (friends) {}
```

Reply:
yeah, I'm just doing it right this moment lolol